testing_and_setup/compass/ocean/global_ocean/scripts/create_E3SM_coupling_files.py

```diff
@@ -0,0 +1,309 @@
+        '-m', mesh_name + '.nc',
+        '-s', scrip_file]
+    run_command(args)
+
+    # make links to output directories
+    os.chdir('../assembled_files_for_upload/input_data/ocn/mpas-o/' + mesh_name)
+    make_link('../../../../../scrip/' + scrip_file, scrip_file)
+
+
+def transects_and_regions(config, date_string):
+
+    # obtain configuration settings
+    mesh_name = config.get('main', 'mesh_name')
+    transect_region_geojson = config.get('transects_and_regions', 'transect_region_geojson')
+
+    maskfile = 'masks_SingleRegionAtlanticWTransportTransects.' + mesh_name + '.nc'
+
+    # make links
+    make_link('../init.nc', mesh_name + '.nc')
+    make_link(transect_region_geojson,
+        'SingleRegionAtlanticWTransportTransects.geojson')
+
+    # check: how does this call MpasMaskCreator?
+    args = ['MpasMaskCreator.x', mesh_name + '.nc', maskfile,
+            '-f', 'SingleRegionAtlanticWTransportTransects.geojson']
+    run_command(args)
+
+    # make links in output directory
+    os.chdir('../assembled_files_for_upload/input_data/ocn/mpas-o/' + mesh_name)
+    make_link('../../../../../transects_and_regions/masks_SingleRegionAtlanticWTransportTransects.' + mesh_name + '.nc',
+             'masks_SingleRegionAtlanticWTransportTransects.' + mesh_name + '.nc')
+
+
+def mapping_Gcase(config, date_string):
+
+    # obtain configuration settings
+    mesh_name = config.get('main', 'mesh_name')
+    atm_scrip_tag = 'T62_040121'
+    atm_scrip_path = config.get('mapping_Gcase', 'atm_scrip_path')
+
+    # make links
+    ocn_scrip_file = 'ocean.' + mesh_name + '.scrip.' + date_string + '.nc'
+    make_link('../scrip/ocean.' + mesh_name + '.scrip.' + date_string + '.nc',
+              ocn_scrip_file)
+    atm_scrip_file = atm_scrip_tag + '.nc'
+    make_link(atm_scrip_path + '/' + atm_scrip_file, atm_scrip_file)
+
+    # execute commands
+    try:
+        method = ['conserve','bilinear','patch']
+        short = ['aave','blin','patc']
+        for j in range(len(method)):
+
+            # Ocean to atmosphere
+            args = ['mpirun', '-n', '1', 'ESMF_RegridWeightGen',
+                    '--method', method[j],
+                    '--source', ocn_scrip_file,
+                    '--destination', atm_scrip_file,
+                    '--weight', 'map_' + mesh_name + '_TO_' + atm_scrip_tag + '_'
+                       + short[j] + '.' + date_string + '.nc',
+                    '--ignore_unmapped']
+            run_command(args)
+
+            # Atmosphere to ocean
+            args = ['mpirun', '-n', '1', 'ESMF_RegridWeightGen',
+                    '--method', method[j],
+                    '--source', atm_scrip_file,
+                    '--destination', ocn_scrip_file,
+                    '--weight', 'map_' + atm_scrip_tag + '_TO_' + mesh_name + '_'
+                       + short[j] + '.' + date_string + '.nc',
+                    '--ignore_unmapped']
+            run_command(args)
```

Comment:
These ESMF calls are likely to be impractically slow on one MPI task, particularly for conservative remapping, for meshes of a realistic size.  On the other hand, I'm not sure the MPI built into ESMF from conda-forge works properly on IC machines.  This is a big issue and I don't know how to solve it!

Reply:
It works on grizzly with e3sm_unified. For Southern Ocean mesh with 720K cells, it takes:
- 26s with -n 1
- 8.3s with -n 36

not too bad either way. I added a flag and set 36 as the default. If other systems have trouble, people can change it back to 1.

Reply:
Oh, that's very exciting!  This probably won't work anywhere on multiple nodes.  And it won't work at all on more than one core on Cori but it's super helpful to know that it works some places!  I'm working on getting `mbtempest` with MPI working in MPAS-Analysis and it seems like this should be useful at least on some machines.